crates/node/core/src/args/pruning.rs

```diff
@@ -3,48 +3,248 @@
+            config.segments.account_history = Some(mode);
+        }
+        if let Some(mode) = self.storage_history_prune_mode() {
+            config.segments.storage_history = Some(mode);
+        }
+
+        if !self.receipts_log_filter.is_empty() {
+            match self.parse_receipts_log_filter() {
+                Ok(filter_config) => config.segments.receipts_log_filter = filter_config,
+                Err(e) => {
+                    // TODO (garwah): Confirm how this should be handled.
+                    eprintln!("Error parsing receipts log filter: {}", e);
+                }
+            }
+        }
+
+        Some(config)
+    }
+    const fn sender_recovery_prune_mode(&self) -> Option<PruneMode> {
+        if self.sender_recovery_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.sender_recovery_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.sender_recovery_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    const fn transaction_lookup_prune_mode(&self) -> Option<PruneMode> {
+        if self.transaction_lookup_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.transaction_lookup_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.transaction_lookup_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    const fn receipts_prune_mode(&self) -> Option<PruneMode> {
+        if self.receipts_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.receipts_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.receipts_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    const fn account_history_prune_mode(&self) -> Option<PruneMode> {
+        if self.account_history_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.account_history_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.account_history_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    const fn storage_history_prune_mode(&self) -> Option<PruneMode> {
+        if self.storage_history_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.storage_history_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.storage_history_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    fn parse_receipts_log_filter(&self) -> Result<ReceiptsLogPruneConfig, String> {
```

Comment:
```suggestion
    fn parse_receipts_log_filter(&self) -> Result<ReceiptsLogPruneConfig, ReceiptsLogError> {
```

needs proper error handling, i.e. make an enum
```rust
enum ReceiptsLogError {
    InvalidFilterFormat(Vec<String>),
    InvalidAddress(Address),
    ...
}
```